src/login/logind-session.c

```diff
@@ -529,6 +534,16 @@ int session_load(Session *s) {
                         s->type = t;
         }
 
+        if (original_type) {
+                SessionType ot;
+
+                ot = session_type_from_string(original_type);
+                if (ot >= 0)
+                        s->original_type = ot;
+        } else {
+                s->original_type = s->type;
+        }
+
```

Comment:
according to our coding style we don't place {} around single line if/else blocks...

can you also add a comment to the code here, that says this is pre-v246 compat?